Caboodle/Sms/Sms.shared.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Sms
+    {
+        public static bool CanSendSms => GetCanSendSms();
+
+        public static bool CanSendSmsInBackground => GetCanSendSmsInBackground();
+
+        public static void SendSms(string recipient = null, string message = null)
+        {
+            if (recipient == null)
+                throw new ArgumentNullException(nameof(recipient));
+
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+
+            if (CanSendSms)
+                PlatformSendSms(recipient, message);
+        }
+
+        public static void SendSmsInBackground(string recipient, string message = null)
+        {
+            if (recipient == null)
+                throw new ArgumentNullException(nameof(recipient));
+
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+
+            if (CanSendSmsInBackground)
+            {
+                PlatformSendSmsInBackground(recipient, message);
+            }
```

Comment:
I think this should throw if it can't send, not fail silently.

Reply:
Already throwing on platform code.

Reply:
What I mean is that this should not be guarded by the `if`. If I call this on iOS, it will just do nothing and I will never know.